client/components/data/query-site-stats/index.jsx

```diff
@@ -36,6 +36,7 @@ class QuerySiteStats extends Component {
 			return;
 		}
 
+		console.log( 'requesting', siteId, statType, query );
```

Comment:
I think this is a good use case for `debug`. Seems like we're using it for other query components already:

https://github.com/Automattic/wp-calypso/blob/472a10bdefcce5bf500898358a314bdc50ec848e/client/components/data/query-site-purchases/index.jsx#L22

It will allow you to keep the logging statement, but only make it visible to folks who want to see it, without cluttering the console for everyone 😉